### PR TITLE
Support mirrors for TIM-VX and others 3rdparty components in opencv_contrib

### PR DESCRIPTION
**Pull request by fengyuentau**
### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or another license that is incompatible with OpenCV
- [x] The PR is proposed to the proper branch
- [ ] There is a reference to the original bug report and related work
- [ ] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [x] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

File: cmake/mirrors/gitcode.cmake [hunk 1]
```diff
@@ -19,11 +30,11 @@ ocv_update(ADE_PKG_MD5_ORIGINAL b624b995ec9c439cbc2e9e6ee940d3a2) # same as ade_
 # 1. Extract repo owner and repo name from DL_URL.
 # 2. Put repo owner and repo name into the placeholders of new DL_URL.
 #
-macro(ocv_download_url_gitcode_usercontent)
+macro(ocv_download_url_gitcode_usercontent OWNER)
   string(REPLACE "/" ";" DL_URL_split ${DL_URL})
   list(GET DL_URL_split 5 __COMMIT_ID)
   list(GET DL_URL_split 6 __PKG_NAME)
-  set(DL_URL "https://gitcode.net/opencv/opencv_3rdparty/-/raw/${__COMMIT_ID}/${__PKG_NAME}/")
+  set(DL_URL "https://gitcode.net/${OWNER}/opencv_3rdparty/-/raw/${__COMMIT_ID}/${__PKG_NAME}/")
```
Comment (asmorkalov) [inline cmake/mirrors/gitcode.cmake:37]: Both Gitcode and Github branches works for me well. But I see issues with double slashes in Gitcode links like this: `https://gitcode.net/mirrors/WeChatCV/opencv_3rdparty/-/raw/a8b69ccc738421293254aec5ddb38bd523503252//detect.caffemodel`. It does not happen in default (github) configuration.

Comment (fengyuentau) [reply cmake/mirrors/gitcode.cmake]: Fixed.

File: cmake/mirrors/gitcode.cmake [hunk 2]
```diff
@@ -34,7 +34,11 @@ macro(ocv_download_url_gitcode_usercontent OWNER)
   string(REPLACE "/" ";" DL_URL_split ${DL_URL})
   list(GET DL_URL_split 5 __COMMIT_ID)
   list(GET DL_URL_split 6 __PKG_NAME)
-  set(DL_URL "https://gitcode.net/${OWNER}/opencv_3rdparty/-/raw/${__COMMIT_ID}/${__PKG_NAME}/")
+  if(__PKG_NAME)
+    set(DL_URL "https://gitcode.net/${OWNER}/opencv_3rdparty/-/raw/${__COMMIT_ID}/${__PKG_NAME}/")
```
Comment (asmorkalov) [inline cmake/mirrors/gitcode.cmake:38]: I propose something like this to not duplicate URLs and logic:
```
set(DL_URL "https://gitcode.net/${OWNER}/opencv_3rdparty/-/raw/${__COMMIT_ID}/")
if(__PKG_NAME)
  set(DL_URL "${DL_URL}${__PKG_NAME}/")
endif()
```

Comment (fengyuentau) [reply cmake/mirrors/gitcode.cmake]: Thanks for the suggestion! Code has been updated to your suggestion.

**Review by asmorkalov — APPROVED** [review]

**Pull request merged**